pytensor/tensor/slinalg.py

```diff
@@ -772,7 +764,7 @@ def _direct_solve_discrete_lyapunov(A: "TensorLike", Q: "TensorLike") -> TensorV
 
 def solve_discrete_lyapunov(
     A: "TensorLike", Q: "TensorLike", method: Literal["direct", "bilinear"] = "direct"
-) -> TensorVariable:
+) -> Variable:
```

Comment:
Can't you add a `typing.cast` on the returned variable?

Reply:
Which one is ultimately correct?

Reply:
I just changed it back, do you think it's worth wrestling with `typing.cast`? It's not done anywhere else in `slinalg`

Reply:
If adding it makes mypy happy yes. A proper solution would be to have `make_node` have a return type hint that specifies what kind of Apply is returned (I have no idea if that's possible atm)

Reply:
A function like this:

```
def(x: TensorVariable) -> TensorVariable:
    return typing.cast(SomeApplyNode(x), TensorVariable)
```

Makes the static checker happy, but add two complaints from mypy:

1. error: Cast target is not a type
2. error: Returning Any from function declared to return "TensorVariable[Any, Any]"

So it seems to be 6 of one, half-dozen of the other. Is there no pure TensorType anywhere in the codebase that should be used as a return type hint for functions? `TensorType` is taken (it's a class, not a Type), and there's `_TensorTypeType`, which is a class, but it's hideous for the name and the leading underscore.

Reply:
Aren't you putting the arguments to cast in the wrong order?

Reply:
> Other functions in slinalg just don't have return type hints, so I can just do that for now I guess.

You can definitely ignore but would be nice to start improving things eventually.

Reply:
Well that's embarrassing